src/System.Net.Http/tests/UnitTests/StreamToStreamCopyTest.cs

```diff
@@ -0,0 +1,351 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Threading.Tasks;
+using Xunit;
+
+namespace System.Net.Http.Tests
+{
+    public class StreamToStreamCopyTest
+    {
+        [Theory]
+        [MemberData(nameof(TwoBooleansWithAdditionalArg), new object[] { new object[] { 256, 8192 } })]
```

Comment:
What about a non-aligned value?  8231 is a nice prime number, so guaranteed to be a partial read for anything other than a buffersize of 8231.